engine/src/main/java/org/hibernate/validator/internal/cfg/context/TypeConstraintMappingContextImpl.java

```diff
@@ -149,7 +150,7 @@ public MethodConstraintMappingContext method(String name, Class<?>... parameterT
 			throw LOG.getBeanDoesNotContainMethodException( beanClass, name, parameterTypes );
 		}
 
-		JavaBeanExecutable<?> javaBeanExecutable = JavaBeanExecutable.of( method );
+		JavaBeanMethod javaBeanExecutable = (JavaBeanMethod) JavaBeanExecutable.of( method );
```

Comment:
Mmmh, let's instantiate a `JavaBeanMethod` directly with the constructor instead of casting then :)

And let's call the variable `javaBeanMethod`.

Reply:
as method here can be either method or getter we couldn't just use a constructor hence I've added an overloaded version of `JavaBeanExecutable#of()` one takes `Method` and returns `JavaBeanMethod` which might also be a getter, and the other one is the one that we already have that can also handle constructors.